p2-maven-plugin/src/main/java/org/eclipse/tycho/p2maven/InstallableUnitGenerator.java

```diff
@@ -343,16 +344,37 @@ public synchronized Collection<IInstallableUnit> getUnits(Artifact artifact) {
 						} catch (IOException e) {
 							// can't determine the type then...
 						}
-						if (isBundle) {
-							List<IPublisherAction> actions = getPublisherActions(PackagingType.TYPE_ECLIPSE_PLUGIN,
-									file, artifact.getVersion(), artifact.getArtifactId());
-							return units = publisher.publishMetadata(actions);
-						}
-						if (isFeature) {
-							List<IPublisherAction> actions = getPublisherActions(PackagingType.TYPE_ECLIPSE_FEATURE,
-									file, artifact.getVersion(), artifact.getArtifactId());
-							return units = publisher.publishMetadata(actions);
+					}
+
+					// TychoReactorReader or another custom WorkspaceReader might have returned
+					// a "dummy" file for the project's artifact if the project is in the reactor.
+					// Usually that file is the project's base directory
+					if (file.isDirectory()) {
+						final File manifestFile = new File(file, "META-INF/MANIFEST.MF");
```

Comment:
If you want such behavior you should simply use the `eclipse-plugin` / `eclipse-feature` packaging for that project.